credentials/local/local.go

```diff
@@ -0,0 +1,104 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package local includes implementation of local credentials.
+// The local credential should be used in either a local TCP or UDS connection.
```

Comment:
How about:

`Local credentials do not provide integrity or privacy protections, but will fail unless a local TCP or UDS connection is used`
?

Maybe also something about reporting the security level based on the connection type.

Reply:
For UDS connection, we assume it provides both privacy and integrity protections.